### ISSUE TITLE
Extend locators with absAt(position)

### ISSUE THREAD

**Issue by nitschSB** [Codeception/CodeceptJS#1670]
#### What are you trying to achieve?

I want to do something with the first element that matches the locator.

#### What do you get instead?

I get an array of all matches.

```js
I.amOnPage('https://www.google.com/search?q=test')
I.grabTextFrom(locate(h3).first())
```

#### Possible Solution
Extend the LocatorAPI (e.g. veryFirst() or absFirst())

```js
  absAt(position) {
    if (position < 0) {
      position++; // -1 points to the last element
      position = `last()-${Math.abs(position)}`;
    }
    if (position === 0) {
      throw new Error('0 is not valid element position. XPath expects first element to have index 1');
    }
    // Important difference brackets around the first %s
    const xpath = sprintf('(%s)[position()=%s]', this.toXPath(), position);
    return new Locator({ xpath });
  }

  absFirst() {
    return this.absAt(1);
  }

  absLast() {
    return this.absAt(-1);
  }
```

### Details

* CodeceptJS version: 2.1.1
* NodeJS Version:  does not matter
* Operating System: does not matter
* puppeteer: 1.11.0
* Configuration file: does not matter

**Comment by heathmills** [comment]
Have you tried this again recently? I ran nearly the same code that you did above (used GrabTextFrom instead of GrabText) and my response was 

`I grab text from {xpath: (.//h3)[position()=1]}`
`Speedtest by Ookla - The Global Broadband Speed Test`

CodeceptJS version: 2.3.1
puppeteer: 1.19.0

**Comment by nitschSB** [comment]
with 2.3.0 my proposed change (round brackets) was released (see #1829) but sadly with a breaking change in the locator api.
